File: app/src/main/java/com/whzl/mengbi/ui/activity/me/NobilityFragment.java
package com.whzl.mengbi.ui.activity.me;

import android.widget.RadioButton;
import android.widget.RadioGroup;

import com.whzl.mengbi.R;
import com.whzl.mengbi.ui.fragment.base.BaseFragment;

import butterknife.BindView;

/**
 * @author niko
 * @date 2018/9/19
 */
public class NobilityFragment extends BaseFragment {
    @BindView(R.id.rb_qingtong)
    RadioButton rbQingtong;
    @BindView(R.id.rb_baiyin)
    RadioButton rbBaiyin;
    @BindView(R.id.rb_huangjin)
    RadioButton rbHuangjin;
    @BindView(R.id.rb_bojin)
    RadioButton rbBojin;
    @BindView(R.id.rb_zuanshi)
    RadioButton rbZuanshi;
    @BindView(R.id.rb_xingyao)
    RadioButton rbXingyao;
    @BindView(R.id.rb_wangzhe)
    RadioButton rbWangzhe;
    @BindView(R.id.rg_nobility)
    RadioGroup rgNobility;

    @Override
    public int getLayoutId() {
        return R.layout.fragment_nobility_shop;
    }

    @Override
    public void init() {
        rbQingtong.setChecked(true);
        rgNobility.setOnCheckedChangeListener((group, checkedId) -> {
            switch (checkedId) {
                case R.id.rb_qingtong:
                    break;
                case R.id.rb_baiyin:
                    break;
                case R.id.rb_huangjin:
                    break;
                case R.id.rb_bojin:
                    break;
                case R.id.rb_zuanshi:
                    break;
                case R.id.rb_xingyao:
                    break;
                case R.id.rb_wangzhe:
                    break;
            }
        });
    }

}

File: app/src/main/java/com/whzl/mengbi/ui/activity/me/NobilityFragment.kt
package com.whzl.mengbi.ui.activity.me

import android.support.v4.app.Fragment
import android.support.v4.view.ViewPager
import butterknife.BindView
import com.whzl.mengbi.R
import com.whzl.mengbi.contract.BasePresenter
import com.whzl.mengbi.contract.BaseView
import com.whzl.mengbi.ui.adapter.FragmentPagerAdaper
import com.whzl.mengbi.ui.fragment.base.BaseFragment
import com.whzl.mengbi.ui.widget.tablayout.TabLayout
import com.whzl.mengbi.util.UIUtil

/**
 *
 * @author nobody
 * @date 2019/3/21
 */
class NobilityFragment : BaseFragment<BasePresenter<BaseView>>() {

    var viewPager: ViewPager? = null
    var tabLayout: TabLayout? = null

    override fun getLayoutId(): Int {
        return R.layout.fragment_nobility
    }

    override fun init() {
        viewPager = activity?.findViewById(R.id.vp_nobility)
        tabLayout = activity?.findViewById(R.id.tablayout_nobility)
        initVp()
    }

    private fun initVp() {
        val titles = arrayListOf("青铜", "白银", "黄金", "铂金", "钻石", "星耀", "王者", "传说")
        val fragments = arrayListOf<Fragment>(
                NobilitySortFragment.newInstance(),
                NobilitySortFragment.newInstance(),
                NobilitySortFragment.newInstance(),
                NobilitySortFragment.newInstance(),
                NobilitySortFragment.newInstance(),
                NobilitySortFragment.newInstance(),
                NobilitySortFragment.newInstance(),
                NobilitySortFragment.newInstance()
        )
        tabLayout?.selectedTabIndicatorWidth = UIUtil.dip2px(activity, 23f)
        tabLayout?.isNeedSwitchAnimation = true
        val adapter = FragmentPagerAdaper(childFragmentManager, fragments, titles)
        viewPager?.adapter = adapter
        tabLayout?.setupWithViewPager(viewPager)
    }
}